src/plugins/str.rs

```diff
@@ -148,6 +168,25 @@ impl Plugin for Str {
         if args.has("to-int") {
             self.for_to_int();
         }
+        if args.has("substring") {
+            /*
+               The intent is to get the 'substring' argument (assuming 'substring' is the tag and 'item' holds the value
+               passed in as 'substring', and save it to the self.params listing as a similarly-shaped Tagged<String>
+             */
+
+            if let Some(position) = args.get("substring") {
+                match position {
+                    Tagged {
+                        item: Value::Primitive(Primitive::String(s)),
+                        tag,
+                    } => {
+                        self.params.push(s.clone().tagged(tag));
+                    }
+                    _ => {}
+                }
+            }
+            self.for_substring();
```

Comment:
once you get the param figured out, then you could pass it as an argument to your `for_substring` call, and then build the enum value there